src/FakeItEasy/Core/FakeManager.cs

```diff
@@ -49,6 +49,7 @@ internal FakeManager(Type fakeObjectType, object proxy, string? fakeObjectName)
             this.FakeObjectName = fakeObjectName;
 
             this.eventRule = new EventRule(this);
+            this.EventCallHandler = new EventCallHandler(this);

```

Comment:
It's a small thing, but we could avoid creating this until/unless we need it. Perhaps many Fakes won't have to handle events. (I know, this is an annoying request, but it's the kind of thing we shaved off during the performance improvements efforts. Or we could just not care that much about performance.)

Reply:
While making this change, I realized that `EventCallHandler` isn't thread-safe (`EventRule` wasn't either). We just use a `Dictionary<K,V>` without any kind of locking. We should probably address this (in a later PR, since this one is big enough already).